smartctl.go

```diff
@@ -60,7 +60,7 @@ func NewSMARTctl(logger log.Logger, json gjson.Result, ch chan<- prometheus.Metr
 		json:   json,
 		logger: logger,
 		device: SMARTDevice{
-			device:     strings.TrimPrefix(strings.TrimSpace(json.Get("device.name").String()), "/dev/"),
+			device:     json.Get("device.info_name").String(),
```

Comment:
This is going to cause a label change when upgrading. Previously it would have shown up as `sda`, but with the PR, it'll show up as `/dev/sda`.

type & protocol should ALSO be represented in the labels, carefully.

Reply:
`/dev` was trimmed in #85 to match `node_exporter` & `netdata` labes. And this is right decision

Seems this code from year '22

Reply:
it's because if you use raid devices has names like **/dev/bus/0 megaraid,0** **/dev/bus/0 megaraid,1** etc.
and if trimming it - we get only one variant **bus/0**

Reply:
It turns out that you need to add the appropriate condition, otherwise you will break already running productions with this change

Reply:
It's your choice.
If you don't need to monitoring a raid disks - just still use of old version of exporter

Reply:
The lack of this ability is keeping many of us from using this exporter at all.

Reply:
> The lack of this ability is keeping many of us from using this exporter at all.

If you not interesting to use this feature - ok, I will use fork of this exporter.
But I found this exporter and saw what it doesn't cover my needs and I trying to help.

Reply:
> It's your choice. If you don't need to monitoring a raid disks - just still use of old version of exporter

The development and integrations don't work in wonderland. In real world you should think about compatibility

Reply:
Ok? why you didn't do that?
If this code is a great problem - how you fix this:
scan shows this
```
/dev/sda -d scsi # /dev/sda, SCSI device
/dev/sdb -d scsi # /dev/sdb, SCSI device
/dev/bus/0 -d megaraid,0 # /dev/bus/0 [megaraid_disk_00], SCSI device
/dev/bus/0 -d megaraid,1 # /dev/bus/0 [megaraid_disk_01], SCSI device
/dev/bus/0 -d megaraid,3 # /dev/bus/0 [megaraid_disk_03], SCSI device
```
and exporter shows this
```
smartctl_device{ata_additional_product_id="DELL(tm)",ata_version="ACS-3 T13/2161-D revision 5",device="sda"}
smartctl_device{ata_additional_product_id="DELL(tm)",ata_version="ACS-3 T13/2161-D revision 5",device="sdb"}
smartctl_device{ata_additional_product_id="DELL(tm)",ata_version="ACS-3 T13/2161-D revision 5",device="bus/0"}
smartctl_device{ata_additional_product_id="DELL(tm)",ata_version="ACS-3 T13/2161-D revision 5",device="bus/0"}
smartctl_device{ata_additional_product_id="DELL(tm)",ata_version="ACS-3 T13/2161-D revision 5",device="bus/0"}
```
?
And what if you have system with little bit more disks?

Reply:
> Ok? why you didn't do that?
> If this code is a great problem - how you fix this

If I were the author of this change, I would use a condition. Here it seems an obvious opportunity to use `megaraid_disk_01` as the device name, because it is unique

Reply:
If you use two raid cards?

Reply:
It will be nice, if you share your `smartctl --scan --json` output

Reply:
```
# /opt/MegaRAID/storcli/storcli64 /call show | grep Name
Product Name = PERC H330 Mini
Driver Name = megaraid_sas
DG/VD TYPE  State Access Consist Cache Cac sCC       Size Name
Product Name = PERC H830 Adapter
Driver Name = megaraid_sas
DG/VD TYPE  State Access Consist Cache Cac sCC       Size Name


# /opt/index/sbin/smartctl --scan
/dev/sda -d scsi # /dev/sda, SCSI device
/dev/sdb -d scsi # /dev/sdb, SCSI device
/dev/sdc -d scsi # /dev/sdc, SCSI device
/dev/sdd -d scsi # /dev/sdd, SCSI device
/dev/sde -d scsi # /dev/sde, SCSI device
/dev/sdf -d scsi # /dev/sdf, SCSI device
/dev/bus/0 -d megaraid,0 # /dev/bus/0 [megaraid_disk_00], SCSI device
/dev/bus/0 -d megaraid,1 # /dev/bus/0 [megaraid_disk_01], SCSI device
/dev/bus/11 -d megaraid,2 # /dev/bus/11 [megaraid_disk_02], SCSI device
/dev/bus/11 -d megaraid,3 # /dev/bus/11 [megaraid_disk_03], SCSI device
/dev/bus/11 -d megaraid,4 # /dev/bus/11 [megaraid_disk_04], SCSI device
/dev/bus/11 -d megaraid,5 # /dev/bus/11 [megaraid_disk_05], SCSI device
/dev/bus/11 -d megaraid,6 # /dev/bus/11 [megaraid_disk_06], SCSI device
/dev/bus/11 -d megaraid,7 # /dev/bus/11 [megaraid_disk_07], SCSI device
/dev/bus/11 -d megaraid,8 # /dev/bus/11 [megaraid_disk_08], SCSI device
/dev/bus/11 -d megaraid,9 # /dev/bus/11 [megaraid_disk_09], SCSI device
/dev/bus/11 -d megaraid,10 # /dev/bus/11 [megaraid_disk_10], SCSI device
/dev/bus/11 -d megaraid,11 # /dev/bus/11 [megaraid_disk_11], SCSI device
/dev/bus/11 -d megaraid,12 # /dev/bus/11 [megaraid_disk_12], SCSI device
/dev/bus/11 -d megaraid,13 # /dev/bus/11 [megaraid_disk_13], SCSI device
/dev/bus/11 -d megaraid,14 # /dev/bus/11 [megaraid_disk_14], SCSI device
/dev/bus/11 -d megaraid,15 # /dev/bus/11 [megaraid_disk_15], SCSI device
/dev/bus/11 -d megaraid,16 # /dev/bus/11 [megaraid_disk_16], SCSI device
/dev/bus/11 -d megaraid,17 # /dev/bus/11 [megaraid_disk_17], SCSI device
/dev/bus/11 -d megaraid,18 # /dev/bus/11 [megaraid_disk_18], SCSI device
/dev/bus/11 -d megaraid,19 # /dev/bus/11 [megaraid_disk_19], SCSI device
/dev/bus/11 -d megaraid,20 # /dev/bus/11 [megaraid_disk_20], SCSI device
/dev/bus/11 -d megaraid,21 # /dev/bus/11 [megaraid_disk_21], SCSI device
/dev/bus/11 -d megaraid,22 # /dev/bus/11 [megaraid_disk_22], SCSI device
/dev/bus/11 -d megaraid,23 # /dev/bus/11 [megaraid_disk_23], SCSI device
/dev/bus/11 -d megaraid,24 # /dev/bus/11 [megaraid_disk_24], SCSI device
/dev/bus/11 -d megaraid,25 # /dev/bus/11 [megaraid_disk_25], SCSI device
/dev/bus/11 -d megaraid,26 # /dev/bus/11 [megaraid_disk_26], SCSI device
/dev/bus/11 -d megaraid,28 # /dev/bus/11 [megaraid_disk_28], SCSI device
/dev/bus/11 -d megaraid,29 # /dev/bus/11 [megaraid_disk_29], SCSI device
/dev/bus/11 -d megaraid,30 # /dev/bus/11 [megaraid_disk_30], SCSI device
/dev/bus/11 -d megaraid,31 # /dev/bus/11 [megaraid_disk_31], SCSI device
/dev/bus/11 -d megaraid,32 # /dev/bus/11 [megaraid_disk_32], SCSI device
/dev/bus/11 -d megaraid,33 # /dev/bus/11 [megaraid_disk_33], SCSI device
/dev/bus/11 -d megaraid,34 # /dev/bus/11 [megaraid_disk_34], SCSI device
/dev/bus/11 -d megaraid,35 # /dev/bus/11 [megaraid_disk_35], SCSI device
/dev/bus/11 -d megaraid,36 # /dev/bus/11 [megaraid_disk_36], SCSI device
/dev/bus/11 -d megaraid,37 # /dev/bus/11 [megaraid_disk_37], SCSI device
/dev/bus/11 -d megaraid,38 # /dev/bus/11 [megaraid_disk_38], SCSI device
/dev/bus/11 -d megaraid,39 # /dev/bus/11 [megaraid_disk_39], SCSI device
/dev/bus/11 -d megaraid,41 # /dev/bus/11 [megaraid_disk_41], SCSI device
/dev/bus/11 -d megaraid,42 # /dev/bus/11 [megaraid_disk_42], SCSI device
/dev/bus/11 -d megaraid,43 # /dev/bus/11 [megaraid_disk_43], SCSI device
/dev/bus/11 -d megaraid,44 # /dev/bus/11 [megaraid_disk_44], SCSI device
/dev/bus/11 -d megaraid,45 # /dev/bus/11 [megaraid_disk_45], SCSI device
/dev/bus/11 -d megaraid,46 # /dev/bus/11 [megaraid_disk_46], SCSI device
/dev/bus/11 -d megaraid,47 # /dev/bus/11 [megaraid_disk_47], SCSI device
/dev/bus/11 -d megaraid,48 # /dev/bus/11 [megaraid_disk_48], SCSI device
/dev/bus/11 -d megaraid,50 # /dev/bus/11 [megaraid_disk_50], SCSI device
/dev/bus/11 -d megaraid,51 # /dev/bus/11 [megaraid_disk_51], SCSI device
/dev/bus/11 -d megaraid,52 # /dev/bus/11 [megaraid_disk_52], SCSI device
/dev/bus/11 -d megaraid,53 # /dev/bus/11 [megaraid_disk_53], SCSI device
/dev/bus/11 -d megaraid,54 # /dev/bus/11 [megaraid_disk_54], SCSI device
/dev/bus/11 -d megaraid,55 # /dev/bus/11 [megaraid_disk_55], SCSI device
/dev/bus/11 -d megaraid,56 # /dev/bus/11 [megaraid_disk_56], SCSI device
/dev/bus/11 -d megaraid,57 # /dev/bus/11 [megaraid_disk_57], SCSI device
/dev/bus/11 -d megaraid,58 # /dev/bus/11 [megaraid_disk_58], SCSI device
/dev/bus/11 -d megaraid,59 # /dev/bus/11 [megaraid_disk_59], SCSI device
/dev/bus/11 -d megaraid,60 # /dev/bus/11 [megaraid_disk_60], SCSI device
/dev/bus/11 -d megaraid,61 # /dev/bus/11 [megaraid_disk_61], SCSI device
/dev/bus/11 -d megaraid,62 # /dev/bus/11 [megaraid_disk_62], SCSI device
/dev/bus/11 -d megaraid,63 # /dev/bus/11 [megaraid_disk_63], SCSI device
/dev/bus/11 -d megaraid,65 # /dev/bus/11 [megaraid_disk_65], SCSI device

```

Reply:
Thanks, today I will make changes

Reply:
Good call to handle multiple HBAs - I've seen multiple tools that either only handle the first, or get very confused glomming multiple together.

What makes it even more complicated is that a given system may have any combo of:

* RAID HBA drives that are part of a VD or DHS/GHS
* RAID HBA drives that are passed through directly (JBOD)
* RAID HBA drives that are not passed through because the HBA is downrev or pissy -- e.g. Dell's H740P.  This category is of least concern IMHO
* Non-RAID HBA drives that are passed through - this includes both AIC cards and onboard / chipset SATA
*

Reply:
Implemented

Reply:
Reverted device param with improvemets:
now you can specify any name of device like:
- sda
- sg0
- nvme0n1p2
- megaraid_disk_01
- etc.